draft-ietf-tls-esni.md

```diff
@@ -986,6 +988,24 @@ Offering a GREASE extension is not considered offering an encrypted ClientHello
 for purposes of requirements in {{real-ech}}. In particular, the client
 MAY offer to resume sessions established without ECH.
 
+### Server Greasing
+
+{{config-extensions-iana}} describes a set of Reserved extensions
+which will never be registered. These can be used by servers to
+"grease" the contents of the ECH configuration, as inspired by
+{{?RFC8701}}. This helps ensure clients process ECH extensions
+correctly. When constructing ECH configurations, servers SHOULD
+randomly select from reserved values with the high-order bit
+clear. Correctly-implemented client will ignore those extensions.
+
+The reserved values with the high-order bit set are mandatory, as
+defined in {{config-extensions}}. Servers SHOULD randomly select from
+these values and include them in extraneous ECH configurations.
+Correctly-implemented clients will ignore these configurations because
+they do not recognize the mandatory extension.  These extraneous ECH
+configurations SHOULD have invalid keys, and invalid public names,
+ending in ".invalid" (see {{?RFC2606}}).
+
```

Comment:
@martinthomson Here's another try:

```suggestion
they do not recognize the mandatory extension.  Servers SHOULD ensure
that any client using these configurations encounters a warning or error
message.  This can be accomplished by giving the extraneous configurations
distinctive config IDs or public names, and rejecting the TLS connection or
inserting an application-level warning message when these are observed.
```

Reply:
I really like this one.  I do think that your list could include making the configuration broken as well (not with a SHOULD, but listed with your "cam be accomplished" phrasing or similar), but this new stuff works for me.